Add profiler and version of seed_from_disk using it

tallenaz opened the pull request:
- lifted `ruby-prof` out of development gem group, to deploy it
- auto-loaded paths in lib, which made it possible to invoke the profiler in a console
- added a profiler that outputs `FlatPrinterWithLineNumbers` (other output options are available)
- added a profiled version of `seed_from_disk` and specs for it
- should probably add specs for the Profiler class

closes #290

ndushay commented on config/application.rb:
```diff
@@ -13,6 +13,7 @@ module PreservationCatalog
   class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
     config.load_defaults 5.1
+    config.autoload_paths += %W[#{config.root}/lib #{config.root}/lib/audit]
```
if this autoload stays here, it can probably be removed from the Rakefile

tallenaz commented:
- Removed the autoload, as it was just a convenience for developing
- Added specs for the Profiler class from pairing with @jmartin-sul

jmartin-sul commented on lib/profiler.rb:
```diff
@@ -0,0 +1,23 @@
+  end
+
+  def print_results_flat(out_file_id)
+    File.open "log/#{out_file_id}-flat.txt", 'w' do |file|
```
what about a timestamp in the filename, just so profiling runs don't step on each other?  could also append to the file, but i think i'd have a slight pref for timestamp, because then you wouldn't have one indefinitely growing file.

jmartin-sul commented on spec/lib/audit/moab_to_catalog_spec.rb:
```diff
@@ -41,6 +41,20 @@
+  describe "#seed_from_disk_with_profiling" do
+    let(:m2c) { described_class.new }
+
+    it "logs what it's about to do, uses the profiler, and prints profiling results" do
```
seems like the logging part got left off?  or i'm just missing it.  but if it did get left off, it'd be nice to either add it, or fix the test description.  logging would be nice, i'd be in favor of adding it, but also doesn't feel super-necessary, so happy to have it omitted if it doesn't feel worth the effort.

tallenaz commented:
- added a timestamp to the log generated by `print_results_flat`
- changed the test descriptions to omit suggestions that we're checking that the profiler actually logs, agreeing that a test like that would be nice to have, and it doesn't seem super-necessary to spec out.

tallenaz commented:
I'll add a comment that the expectation involving timestamps on the profiler log might need to be loosened.
